src/services/http/patterns.js

```diff
@@ -0,0 +1,136 @@
+'use strict'
+const jwt = require('jsonwebtoken')
+const bcrypt = require('bcryptjs')
+const { crypt } = require('../../config/vars')
+const { responseHandler, errorHandler } = require('../../services/http/utils/response')
+
+/**
+ * Plugin handles the user entity data store
+ */
+module.exports = function plugin (options) {
+  var seneca = this
+
+  /**
+     * Handles creating a new administrator
+     */
+  seneca.add('role:auth,cmd:signup', (msg, done) => {
+    let args = msg.args.body
+    try {
+      bcrypt.genSalt(crypt.SALT_WORK_FACTOR, function (err, salt) {
+        bcrypt.hash(args.password, salt, function (err, hash) {
+          seneca.act('role:storage,cmd:insert', ({ type: 'administrator', input: { username: args.username, password: hash } }), function (err, res) {
+            console.log(res)
+            delete res.password
+            done(err, responseHandler(res))
+          })
+        })
+      })
+    } catch (err) {
+      done(err, responseHandler(err))
+    }
+  })
+
+  /**
+     * this action handles signing payload and returns a valid token for authentication
+     */
+  seneca.add('role:auth,cmd:login', (msg, done) => {
+    let args = msg.args.body
+    let message
+    seneca.act('role:storage,cmd:fetchAdmin', ({ username: args.username }), function (err, admin) {
+      message = {
+        error: 'invalid credentials'
+      }
+      if (admin) {
+        bcrypt.compare(args.password, admin.password, function (err, res) {
+          if (res === true) {
+            let payload = { id: admin._id, username: admin.username }
+            let token = jwt.sign(payload, options.secretOrKey)
+            message = {
+              token: token
+            }
+            done(err, responseHandler(message))
+          } else {
+            done(err, responseHandler(message))
+          }
+        })
+      } else {
+        done(err, responseHandler(message))
+      }
+    })
+  })
+
+  /**
+     * handles returning a user entity
+     */
+  seneca.add('role:user,cmd:load', async (msg, done) => {
```

Comment:
it seems async is not needed here?

Reply:
Yes its not needed , you will note that in some of the other functions I omitted it , the reason is because when I started I had an idea where I could completely encapsulate the actions  , that way the functions could be resolved easily and maintained better instead of having them all in the patterns file , I would then be able to map them accordingly. Also in case I needed to use await this will enable me to go ahead and do that